geode-wan/src/distributedTest/java/org/apache/geode/internal/cache/wan/concurrent/ConcurrentParallelGatewaySenderDUnitTest.java

```diff
@@ -596,32 +620,34 @@ public void testPartitionedParallelPropagationHA() throws Exception {
 
     startSenderInVMs("ln", vm4, vm5, vm6, vm7);
 
-    AsyncInvocation inv1 =
+    AsyncInvocation putsToVM7 =
         vm7.invokeAsync(() -> WANTestBase.doPuts(getTestMethodName() + "_PR", 5000));
 
     vm2.invoke(() -> Awaitility.await().atMost(30000, TimeUnit.MILLISECONDS)
         .untilAsserted(() -> assertEquals(
             "Failure in waiting for at least 10 events to be received by the receiver", true,
             (getRegionSize(getTestMethodName() + "_PR") > 10))));
 
-    AsyncInvocation inv2 = vm4.invokeAsync(() -> WANTestBase.killSender());
+    AsyncInvocation killsSenderFromVM4 = vm4.invokeAsync(() -> WANTestBase.killSender());
 
     int prevRegionSize = vm2.invoke(() -> WANTestBase.getRegionSize(getTestMethodName() + "_PR"));
 
-    AsyncInvocation inv3 =
+    AsyncInvocation putsToVM6 =
         vm6.invokeAsync(() -> WANTestBase.doPuts(getTestMethodName() + "_PR", 10000));
 
-    vm2.invoke(() -> Awaitility.await().atMost(30000, TimeUnit.MILLISECONDS)
-        .untilAsserted(() -> assertEquals(
-            "Failure in waiting for additional 20 events to be received by the receiver ", true,
-            getRegionSize(getTestMethodName() + "_PR") > 20 + prevRegionSize)));
+    AsyncInvocation assertsRegionSizeInVM2 =
+        vm2.invokeAsync(() -> Awaitility.await().atMost(30000, TimeUnit.MILLISECONDS)
+            .untilAsserted(() -> assertEquals(
+                "Failure in waiting for additional 20 events to be received by the receiver ", true,
+                getRegionSize(getTestMethodName() + "_PR") > 20 + prevRegionSize)));
 
-    AsyncInvocation inv4 = vm5.invokeAsync(() -> WANTestBase.killSender());
+    AsyncInvocation killsSenderFromVM5 = vm5.invokeAsync(() -> WANTestBase.killSender());
 
-    inv1.join();
-    inv2.join();
-    inv3.join();
-    inv4.join();
+    putsToVM7.join();
+    killsSenderFromVM4.join();
+    putsToVM6.join();
+    killsSenderFromVM5.join();
+    assertsRegionSizeInVM2.join();
```

Comment:
Good call, Dan. This looked like it had fixed it because join doesn't throw the failure out. I will be reverting this change and I'll try to find a new way to do it.